### PR TITLE
chore: Add error conversion from `InterpreterError`

### PR DESCRIPTION
**Pull request by jfecher**
# Description

## Problem\*

Resolves comment: https://github.com/noir-lang/noir/pull/4884#discussion_r1576646720

## Summary\*

Converts from `InterpreterError` to `CustomDiagnostic` and removes the previous `.unwrap()` placeholder so that interpreter errors are now properly displayed.

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by michaeljklein — APPROVED** [review]
LGTM

File: compiler/noirc_frontend/src/hir/comptime/errors.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+use crate::{hir::def_collector::dc_crate::CompilationError, Type};
 use acvm::FieldElement;
-use noirc_errors::Location;
+use noirc_errors::{CustomDiagnostic, Location};
 
 use super::value::Value;
 
 /// The possible errors that can halt the interpreter.
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (vezenovm) [inline compiler/noirc_frontend/src/hir/comptime/errors.rs:8]:
```suggestion
#[derive(Debug)]
```

Comment (vezenovm) [reply compiler/noirc_frontend/src/hir/comptime/errors.rs:8]: Don't think this is used?

**Review by vezenovm — APPROVED** [review]
Looks good

File: compiler/noirc_frontend/src/hir/comptime/errors.rs (hunk 1 above)
Comment (jfecher) [reply compiler/noirc_frontend/src/hir/comptime/errors.rs]: Looks like it is required by the bound on CompilationError

Comment (vezenovm) [reply compiler/noirc_frontend/src/hir/comptime/errors.rs]: Ah my bad I missed that

**Pull request merged**